### PR TITLE
Refactor backend use methodview phase 1

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Refactor Flask API to use MethodView in lieu of Flask Restful (deprecated).
Moved models to separate package in preparation to separate it into its own component.
Reorganized code structure.
Added lots of endpoint unit tests.
Versioned API at v1.

N.B. Still more to do but this is a first broad pass.

### CODE REVIEW

**Comment by xlorepdarkhelm** [comment]
I am sure you saw this:

> Check warning on line 44 in frontend/src/components/PortfolioDescription/PortfolioDescription.jsx
> 
> GitHub Actions
> / Linting
> Array.prototype.map() expects a value to be returned at the end of arrow function

**Comment by jonnalley** [comment]
> I am sure you saw this:
> 
> > Check warning on line 44 in frontend/src/components/PortfolioDescription/PortfolioDescription.jsx
> > GitHub Actions
> > / Linting
> > Array.prototype.map() expects a value to be returned at the end of arrow function

This was fixed in a previous PR and got taken care of when main was merged into this branch after the comment was made.

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

File: backend/ops_api/ops/resources/auth.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from flask import jsonify
+from flask import request
+from ops.base_views import BaseListAPI
+from ops.utils.auth_views import login
+from ops.utils.auth_views import refresh
+
+
+class AuthLoginAPI(BaseListAPI):
+    def __init__(self, model):
```
[the remark this answers is not in the document]
Comment (johndeange) [reply backend/ops_api/ops/resources/auth.py]: looks like I missed a bunch of type hints - adding more

**Pull request merged**